### ISSUE TITLE
Provide Restart Current Task action

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#11204]
- VSCode Version: 1.5 Insider
- OS Version: Windows

See https://github.com/Microsoft/vscode/issues/6578#issuecomment-243417095 why this is necessary now

**Comment by dbaeumer** [comment]
This is a lot more complicated to implement then I thought. Reason being is that terminating a process is signaled async the the streams of the process close. So if then the new task is already running its state will be cleared. This is an issue with the system in general but since a user can't executed terminate and run fast enough it never shows up. Need to postpone this.

**Comment by dbaeumer** [comment]
Spent my time adoption TS 2.0.3. This has to wait a little.

**Comment by dbaeumer** [comment]
I re-implemented the task execution engine for Jan to use the terminal instead of the output window. Although not the default yet I will wait until the switch is final to provide a restart current task action. With the new engine this needs more thinking since the new engine support executing tasks in parallel.
